native-engine/datafusion-ext-plans/src/shuffle/buffered_data.rs

```diff
@@ -61,10 +63,9 @@ impl BufferedData {
 
     pub fn add_batch(&mut self, batch: RecordBatch, partitioning: &Partitioning) -> Result<()> {
         self.num_rows += batch.num_rows();
-
         let (parts, sorted_batch) = self
             .sort_time
-            .with_timer(|| sort_batch_by_partition_id(batch, partitioning))?;
+            .with_timer(|| sort_batch_by_partition_id(batch, partitioning, self.num_rows))?;
```

Comment:
we can sort before updating `self.num_rows` so that we don't need to substract `batch.num_rows()` in the implementation.